e2e/cypress/tests/support/api/system.d.ts

```diff
@@ -89,15 +89,15 @@ declare namespace Cypress {
         /**
          * Update configuration.
          * See https://api.mattermost.com/#tag/system/paths/~1config/put
-         * @param {AdminConfig} newConfig - new config
+         * @param {Partial<AdminConfig>} newConfig - new config
          * @returns {AdminConfig} `out.config` as `AdminConfig`
          *
          * @example
          *   cy.apiUpdateConfig().then(({config}) => {
          *       // do something with config
          *   });
          */
-        apiUpdateConfig(newConfig: AdminConfig): Chainable<AdminConfig>;
+        apiUpdateConfig(newConfig: Partial<AdminConfig>): Chainable<AdminConfig>;
```

Comment:
This doesn't fully work because we need to use a DeepPartial type. I had trouble with using the DeepPartial util type from Mattermost (summary in the commit message).

Reply:
# Commit Message for Reference
All props in the ApiConfig type are required. And each of child configs
required all props.

I tried to use the DeepPartial utility but that didn't work.
Importing DeepPartial turned the file into a Module file and caused the
functions defined in system.d.ts to disappear from the Cypress namespace

See this GH Issue thread for more information:
https://github.com/microsoft/TypeScript-Handbook/issues/180#issuecomment-196475664

I could potentially combine the system.js & system.d.ts files
into a single ts file.

Submitting this WIP commit for feedback. Thanks.

Reply:
This is now using `DeepPartial` on latest master. Please update.